### ISSUE TITLE
"nix-env -qas" fails due to samba

### ISSUE THREAD

**Issue by giogadi** [NixOS/nixpkgs#6681]
```bash
luis$ nix-env --show-trace -qas                                                           
error: while evaluating the attribute ‘configureFlags’ of the derivation ‘samba-4.2.0’ at "/Users/luis/code/nixpkgs/pkgs/servers/samba/4.x.nix":33:3:
while evaluating ‘optional’ at "/Users/luis/code/nixpkgs/lib/lists.nix":105:20, called from "/Users/luis/code/nixpkgs/pkgs/servers/samba/4.x.nix":104:8:
value is null while a set was expected, at "/Users/luis/code/nixpkgs/pkgs/servers/samba/4.x.nix":104:29
```

I'm using nixpkgs from GitHub on OSX Yosemite. I just did a fresh pull a few minutes ago.

**Comment by vcunat** [comment]
Thanks! I'll wait some time for answer to 936fdc7b6.

**Comment by edolstra** [comment]
Should be fixed in f72ae4da79e15c78cf22eb99a0af6d1816de5e7b.
